inlang/source-code/sdk/src/createNewProject.test.ts

```diff
@@ -0,0 +1,111 @@
+import { describe, it, expect } from "vitest"
+import { createNewProject } from "./createNewProject.js"
+import { mockRepo } from "@lix-js/client"
+import { defaultProjectSettings } from "./defaultProjectSettings.js"
+import { loadProject } from "./loadProject.js"
+import { createMessage } from "./test-utilities/createMessage.js"
+
+function sleep(ms: number) {
+	return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+describe("createNewProject", () => {
+	it("should throw if a path does not end with .inlang", async () => {
+		try {
+			await createNewProject({
+				projectPath: "/boom/ba/da/bang",
+				repo: await mockRepo(),
+			})
+			throw new Error("Expected an error")
+		} catch (e) {
+			expect((e as Error).message).toMatch(
```

Comment:
I would use https://vitest.dev/api/expect#tothrowerror (tip, because i missed it: read till the end it suggest using .rejects for promises)

Reply:
@martin-lysk I prefer the existing pattern even though it's a bit more verbose.

The suggested pattern using `async expect(() => fn()).rejects.toThrowError()` is more consise, but less readable because it requires internal knowledge of `async expect()` and `reject`, and forces the async function to be called without await. e.g.

```ts
it("should throw if a path does not end with .inlang", async () => {
	const repo = await mockRepo()
	const projectPath = "/test/name.inl"
	await expect(() => createNewProject({ projectPath, repo })).rejects.toThrowError(
		'Expected a path ending in "{name}.inlang" but received "/test/name.inl"'
	)
})
```